pkg/didcomm/transport/ws/inbound.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package ws
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/websocket"
+
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
+	commontransport "github.com/hyperledger/aries-framework-go/pkg/didcomm/common/transport"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/transport"
+)
+
+var logger = log.New("aries-framework/ws")
+
+const processFailureErrMsg = "failed to process the message"
+
+// provider contains dependencies for the WS Handler creation and is typically created by using aries.Context()
+type provider interface {
```

Comment:
is it needs to duplicate the same interface here? 

you can simply implement the same interface found here https://github.com/hyperledger/aries-framework-go/blob/master/pkg/didcomm/transport/transport_interface.go#L28

unless you intend to create new functions

Reply:
Valid point, I missed the common interface. [Removed](https://github.com/hyperledger/aries-framework-go/compare/0cf0b7e4209cfb161f30fbe9b94d326c76bf2b0d..862a01f471d674befdfadae8eadbcab750c23edc#diff-a920dee998c6d38ee04585f60a90626bL27).